nik4.py

```diff
@@ -552,6 +576,7 @@ def run(options):
                 if result == 0:
                     for tile in tile_files:
                         os.remove(tile)
+                    write_metadata(bbox, size[0], size[1], transform, options.output, options.wld, options.ozi)
```

Comment:
Since `write_metadata` is at the end of every `if` clause, can it be inserted only once before `if options.output == '-'` line? What check would be necessary to do that?

Reply:
In line 565 (if `options.just_tiles` is `True`), world files are written for each tile because if `--just-tiles` is given, the tiles are not stitched together to one single image and a world file for all tiles would not be useful.

Reply:
Okay, but what about leaving this here and move out others?

Reply:
Oh, I think I see the point. I deduplicated lines 510 and 513.

However, I think it would be useful to write the worldfile only if the rendering (and stitching) was successful.